src/Compilers/Core/Portable/SourceGeneration/IncrementalGeneratorInitializationContext_ForAttributeWithSimpleName.cs

```diff
@@ -0,0 +1,284 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Transactions;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.SourceGeneration;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis;
+
+using Aliases = ArrayBuilder<(string aliasName, string symbolName)>;
+
+public partial struct IncrementalGeneratorInitializationContext
+{
+    private static readonly ObjectPool<Stack<string>> s_stackPool = new(static () => new());
+
+    /// <summary>
+    /// Returns all syntax nodes of that match <paramref name="predicate"/> if that node has an attribute on it that
+    /// could possibly bind to the provided <paramref name="simpleName"/>. <paramref name="simpleName"/> should be the
+    /// simple, non-qualified, name of the attribute, including the <c>Attribute</c> suffix, and not containing any
+    /// generics, containing types, or namespaces.  For example <c>CLSCompliantAttribute</c> for <see
+    /// cref="System.CLSCompliantAttribute"/>.
+    /// <para/> This provider understands <see langword="using"/> (<c>Import</c> in Visual Basic) aliases and will find
+    /// matches even when the attribute references an alias name.  For example, given:
+    /// <code>
+    /// using XAttribute = System.CLSCompliantAttribute;
+    /// [X]
+    /// class C { }
+    /// </code>
+    /// Then
+    /// <c>context.SyntaxProvider.CreateSyntaxProviderForAttribute(nameof(CLSCompliantAttribute), (node, c) => node is ClassDeclarationSyntax)</c>
+    /// will find the <c>C</c> class.
+    /// </summary>
+    internal IncrementalValuesProvider<SyntaxNode> ForAttributeWithSimpleName(
+        string simpleName,
+        Func<SyntaxNode, CancellationToken, bool> predicate)
+    {
+        var syntaxHelper = this.SyntaxHelper;
+
+        // Create a provider that provides (and updates) the global aliases for any particular file when it is edited.
+        var individualFileGlobalAliasesProvider = this.SyntaxProvider.CreateSyntaxProvider(
+            static (n, _) => n is ICompilationUnitSyntax,
+            static (context, _) => getGlobalAliasesInCompilationUnit(context.SyntaxHelper, context.Node)).WithTrackingName("individualFileGlobalAliases_ForAttribute");
```

Comment:
We need the tracking names for our internal testing, but they 'leak' into public as they'll now show up in users tracking results if they use this API. I think thats fine, but I think we should get consensus that we're a) ok with that and b) agree that this is not a public API contract and the nodes can be removed / renamed without it being considered a breaking change.

Reply:
> but I think we should get consensus that we're a) ok with that and b) agree that this is not a public API contract and the nodes can be removed / renamed without it being considered a breaking change.

Totally agreed.  We coudl bring that to API meeting and then doc this explicitly.